docs/auth/server/server.js

```diff
@@ -0,0 +1,81 @@
+const createError = require('http-errors');
+const express = require('express');
+const morgan = require('morgan');
+const mustacheExpress = require('mustache-express');
+const session = require('express-session');
+const http = require('http');
+
+const app = express();
+
+app.engine('mustache', mustacheExpress());
+app.set('view engine', 'mustache');
+app.set('views', __dirname + '/views');
+
+app.use(morgan('dev'));
+app.use(express.json());
+app.use(express.urlencoded({extended: false}));
+
+app.use(session({
+  secret: 'notverysecret',
+  resave: true,
+  saveUninitialized: false,
+}));
+
+app.use(function(req, res, next) {
+  res.locals.session = req.session;
+  next();
+});
+
+function loginRequired(req, res, next) {
```

Comment:
I'm not sure if the aim was the make sure we can handle a more typical/sophisticated setup or to show users it works with a typical setup, but if you're not used to express I think this example would be fairly opaque compared to 

```js
app.get('dashboard', (req, res) => {
  if (!req.session.user) return res.redirect('/login');
  res.render('dashboard')
})

```

similarly I think the full mustache template system might be a bit much for a user just looking for an authentication example to go through, but again I'm not 100% of the motivation behind it so I might be missing context.

Reply:
I'm a weary to make the Express code less idiomatic, especially since even looking at this code is optional to the goal of the docs: to understand how to run LH on auth pages.

I want the pages to serve different content based on session state, as that is a realistic setup. For those that _do_ know how Node/express servers work, using explicit `write` calls to create the document will be confusing, much more than the concept of a server having a template engine would be to those with zero Express experience.

I'm less convinced about the auth middleware (`loginRequired`). It's just used once so it feels OK to just move it into the `/dashboard` route.

Reply:
> similarly I think the full mustache template system might be a bit much for a user just looking for an authentication example to go through

I don't feel strongly, but I do generally agree with @patrickhulce. We want an example real enough that it's clear to users how to adapt to their own use cases, but also not get too far into a particular stack. If I'm not an express dev (maybe I don't use node at all?) how much of that and idiomatic express code matters?

One nice thing is that it's mostly isolated from the test and simple auth files, so some @fileoverview entries (and links at the top of the readme) might be enough in directing those sorts of devs to the right spots to be looking instead of getting weighed down in the server code or mustache templates.

Reply:
I'm not convinced that 
```js
app.get('/', (req, res) => {
  if (req.session.user) {
    res.send(`<html><h1>Hello ${req.session.user}, secrets</h1></html>`)
  } else {
    res.send(`<html><h1>Login first!</h1></html>`)
  }
});
```
is *more* confusing than learning js server-side template engines, but I won't die on this hill either. You did a lot of work to make it more realistic which is great 👏👏👏 , so it's w/e

Reply:
ok i'm sold, i'll remove the mustache too 👨 ➡️ 	 👶